deals/module/deals.go

```diff
@@ -327,22 +327,29 @@ func (m *Module) Watch(ctx context.Context, proposals []cid.Cid) (<-chan deals.S
 	}
 	ch := make(chan deals.StorageDealInfo)
 	go func() {
-		defer close(ch)
+		client, cls, err := m.clientBuilder()
```

Comment:
Goal of these changes is to make watching deals send a result immediately upon subscription instead of waiting `m.pollDuration` amount of time before getting the first event.